pkg/diagnose/deployments.go

```diff
@@ -177,6 +176,9 @@ func checkPodsStatus(k8sClient kubernetes.Interface, namespace string, status re
 	for i := range pods.Items {
 		pod := &pods.Items[i]
 		if pod.Status.Phase != v1.PodRunning {
+			if pod.Labels["type"] == "submariner-diagnose" {
```

Comment:
You should utilize label selectors in `ListOptions` instead (line 170).

You can find examples how to do it in the code.

Also I don't know if type is appropriate, maybe something like `transient: true` to indicate pods that we want to ignore (might be not just the diagnose pods in the future)